### PR TITLE
Change license for executable and test suite to AGPLv3 or later

### PR DESCRIPTION
**Pull request by jbboehr**
The library remains licensed under LGPLv2.1 or later

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -75,6 +75,7 @@ cd handlebars.c
 
 ## License
 
-This project is licensed under the [LGPLv2.1 or later](LICENSE.md).
+The library for this project is licensed under the [LGPLv2.1 or later](LICENSE.md).
+The executable and the test suite are licensed under the [AGPLv3.0 or later](LICENSE-AGPL.md).
 handlebars.js is licensed under the [MIT license](http://opensource.org/licenses/MIT).
```
Comment (bkotsopoulossc) [inline README.md:80]: Hi @jbboehr, is it possible to use a more permissive license like MIT for the project itself?

Comment (jbboehr) [reply README.md:80]: Most likely not. Do you have a use case for which the current licensing scheme is prohibitive?

Comment (bkotsopoulossc) [reply README.md:80]: Yes, to statically link the library without modifications, into a closed-source iOS app

Comment (jbboehr) [reply README.md:80]: Not super familiar with iOS development, but I assume it can't be dynamically linked?

Comment (bkotsopoulossc) [reply README.md:80]: It’s possible in theory but has a lot of tradeoffs like startup time, losing out on LTO, harder to symbolicate crashes, have to be mindful of which symbols to export, etc

Comment (jbboehr) [reply README.md:80]: If you don't mind playing with legal fire (IANAL), I believe I can grant you a static linking exception.

Comment (bkotsopoulossc) [reply README.md:80]: That would be hard to get through legal. If you are willing to allow static linking would you be open to supporting the Mozilla public license?

https://softwareengineering.stackexchange.com/questions/221365/mozilla-public-license-mpl-2-0-vs-lesser-gnu-general-public-license-lgpl-3-0

Comment (jbboehr) [reply README.md:80]: One problem: this project currently has a hard dependency on talloc (really it's only hard dependency), which is LGPL-3.0 licensed.

I have considered the MPL before.

Comment (jbboehr) [reply README.md:80]: In any case, you have my permission to use the library under the terms of [LGPL-3.0-linking-exception](https://spdx.org/licenses/LGPL-3.0-linking-exception.html), but only on iOS.

**Pull request closed without merging**